shark-graph/src/test/java/shark/LongScatterSetTest.kt

```diff
@@ -1,142 +1,137 @@
-    val set = LongScatterSet()
-    runAssertionsForAddOperation(set)
+  @Test fun `new set is empty`() {
+    assertThat(LongScatterSet()).isEmpty()
   }
 
-  @Test fun `verify LongScatterSet#remove operation works correctly`() {
+  @Test fun `adding elements`() {
     val set = LongScatterSet()
-    runAssertionsForRemoveOperation(set)
+
+    TEST_VALUES_LIST.forEach { set.add(it) }
+
+    assertThat(set)
+        .contains(*TEST_VALUES_LIST.toLongArray())
+        .hasSize(TEST_VALUES_LIST.size)
   }
 
-  @Test fun `verify LongScatterSet#release operation works correctly`() {
+  @Test fun `+= operator works as addition`() {
     val set = LongScatterSet()
-    set.add(42)
-    set.release()
+
+    set.add(TEST_VALUE)
 
     assertThat(set)
-        .isEmpty()
-        .doesNotContain(42)
+        .contains(TEST_VALUE)
+        .hasSize(1)
   }
 
-  @Test fun `verify LongScatterSet#ensureCapacity does not break Set when called before those operations`() {
-    val setForAddOperation = LongScatterSet().apply { ensureCapacity(10) }
-    val setForRemoveOperation = LongScatterSet().apply { ensureCapacity(10) }
+  @Test fun `adding element twice doesn't cause duplicates`() {
+    val set = LongScatterSet()
+
+    set.add(TEST_VALUE)
+    set.add(TEST_VALUE)
 
-    // Verify that calling ensureCapacity doesn't affect add/remove operations
-    runAssertionsForAddOperation(setForAddOperation)
-    runAssertionsForRemoveOperation(setForRemoveOperation)
+    assertThat(set)
+        .contains(TEST_VALUE)
+        .hasSize(1)
   }
 
-  @Test fun `verify LongScatterSet#ensureCapacity does not break Set when called after those operations`() {
-    val set = LongScatterSet().apply {
-      add(42)
-      add(10)
-      ensureCapacity(100)
-    }
+  @Test fun `adding elements with matching hash`() {
+    val set = LongScatterSet()
+
+    set += SAME_MIX_PHI_1
+    set += SAME_MIX_PHI_2
 
-    // Verify that calling ensureCapacity didn't change any data in set
     assertThat(set)
-        .contains(10, 42)
+        .contains(SAME_MIX_PHI_1, SAME_MIX_PHI_2)
         .hasSize(2)
   }
 
-  private fun runAssertionsForRemoveOperation(set: LongScatterSet) {
-    require(set.size() == 0) { "Set should be empty in order to run the assertions on it" }
+  @Test fun `removing elements`() {
+    val set = LongScatterSet()
+    TEST_VALUES_LIST.forEach { set.add(it) }
 
-    val testValues = listOf(42, 0, Long.MIN_VALUE, Long.MAX_VALUE, -1)
+    TEST_VALUES_LIST.forEach { set.remove(it) }
 
-    // First, verify removing unique values from an empty set
-    testValues.forEach { value ->
-      set.remove(value)
-      assertThat(set)
-          .isEmpty()
-          .doesNotContain(value)
-    }
+    assertThat(set)
+        .doesNotContain(*TEST_VALUES_LIST.toLongArray())
+        .isEmpty()
+  }
 
-    // Add test values to the set
-    testValues.forEach { set += it }
+  @Test fun `removing from empty set`() {
```

Comment:
Is there any world in which a change to LongScatterSet could break this behavior?

Reply:
Of course! Any breaking change to `hashKey` or misconfiguration of initial values of `keys` or `mask` can cause incorrect result for `existing` (or throw an error)